diff -Nur tmview.orig/MakeFb tmview/MakeFb
--- tmview.orig/MakeFb	Sun Mar  4 14:57:22 2001
+++ tmview/MakeFb	Fri Nov  9 19:48:38 2001
@@ -8,8 +8,10 @@
 LIBS= -lm	 
 #LIBS= -lkpathsea -lm       # when kpathsea is to be used	 
 
-LL=gcc   -L /usr/lib
-CC=gcc -c -O2 -g -Wall -I$(SRC) -I$(SSRC)
+LL=gcc
+CC=gcc -c
+CFLAGS=-O2 -g -Wall
+INCLUDES=-I$(SRC) -I$(SSRC)
 
 OBJDEPEND=$(SSRC)defsfb.h $(SRC)defsgen.h $(SRC)subs.c
 
@@ -34,46 +36,46 @@
 
 
 $(OBJ)tmview.o:  $(SRC)tmview.c $(SRC)help.h $(OBJDEPEND)
-	$(CC) $(SRC)tmview.c -o $(OBJ)tmview.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)tmview.c -o $(OBJ)tmview.o
 
 $(OBJ)readdvi.o:  $(SRC)readdvi.c $(OBJDEPEND)
-	$(CC) $(SRC)readdvi.c -o $(OBJ)readdvi.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)readdvi.c -o $(OBJ)readdvi.o
 
 $(OBJ)readpk.o:  $(SRC)readpk.c $(OBJDEPEND)
-	$(CC) $(SRC)readpk.c -o $(OBJ)readpk.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)readpk.c -o $(OBJ)readpk.o
 
 $(OBJ)readvf.o:  $(SRC)readvf.c $(OBJDEPEND)
-	$(CC) $(SRC)readvf.c -o $(OBJ)readvf.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)readvf.c -o $(OBJ)readvf.o
 
 $(OBJ)readtfm.o:  $(SRC)readtfm.c $(OBJDEPEND)
-	$(CC) $(SRC)readtfm.c -o $(OBJ)readtfm.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)readtfm.c -o $(OBJ)readtfm.o
 
 $(OBJ)drawit.o:  $(SRC)drawit.c $(OBJDEPEND)
-	$(CC) $(SRC)drawit.c -o $(OBJ)drawit.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)drawit.c -o $(OBJ)drawit.o
 
 $(OBJ)halfhyp.o:  $(SRC)halfhyp.c $(OBJDEPEND)
-	$(CC) $(SRC)halfhyp.c -o $(OBJ)halfhyp.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)halfhyp.c -o $(OBJ)halfhyp.o
 
 $(OBJ)epsfile.o:  $(SRC)epsfile.c $(OBJDEPEND)
-	$(CC) $(SRC)epsfile.c -o $(OBJ)epsfile.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)epsfile.c -o $(OBJ)epsfile.o
 
 $(OBJ)colors.o:  $(SRC)colors.c $(OBJDEPEND)
-	$(CC) $(SRC)colors.c -o $(OBJ)colors.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)colors.c -o $(OBJ)colors.o
 
 $(OBJ)writefb.o:  $(SSRC)writefb.c $(OBJDEPEND)
-	$(CC) $(SSRC)writefb.c -o $(OBJ)writefb.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SSRC)writefb.c -o $(OBJ)writefb.o
 
 $(OBJ)subs.o:  $(SRC)subs.c $(OBJDEPEND)
-	$(CC) $(SRC)subs.c -o $(OBJ)subs.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)subs.c -o $(OBJ)subs.o
 
 $(OBJ)bookmks.o:  $(SRC)bookmks.c $(OBJDEPEND)
-	$(CC) $(SRC)bookmks.c -o $(OBJ)bookmks.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)bookmks.c -o $(OBJ)bookmks.o
 
 $(OBJ)rwconf.o:  $(SRC)rwconf.c $(OBJDEPEND)
-	$(CC) $(SRC)rwconf.c -o $(OBJ)rwconf.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)rwconf.c -o $(OBJ)rwconf.o
 
 $(OBJ)globals.o:  $(SRC)globals.c $(OBJDEPEND)
-	$(CC) $(SRC)globals.c -o $(OBJ)globals.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)globals.c -o $(OBJ)globals.o
 clean:
 	rm -f $(OBJ)*.o
 
diff -Nur tmview.orig/MakeLX tmview/MakeLX
--- tmview.orig/MakeLX	Sun Mar  4 14:57:22 2001
+++ tmview/MakeLX	Fri Nov  9 19:43:31 2001
@@ -12,7 +12,9 @@
 LIBS= -L /usr/X11R6/lib -L /usr/X11/lib  -lX11 -lm        
 
 LL=gcc
-CC=gcc -c -O2 -g -Wall -I$(SRC) -I$(SSRC) 
+CC=gcc -c
+CFLAGS=-O2 -g -Wall
+INCLUDES=-I$(SRC) -I$(SSRC)
 
 
 
@@ -37,46 +39,46 @@
 
 
 $(OBJ)tmview.o:  $(SRC)tmview.c $(SRC)help.h $(OBJDEPEND) 
-	$(CC) $(SRC)tmview.c -o $(OBJ)tmview.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)tmview.c -o $(OBJ)tmview.o
 
 $(OBJ)readdvi.o:  $(SRC)readdvi.c $(OBJDEPEND)
-	$(CC) $(SRC)readdvi.c -o $(OBJ)readdvi.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)readdvi.c -o $(OBJ)readdvi.o
 
 $(OBJ)readpk.o:  $(SRC)readpk.c $(OBJDEPEND)
-	$(CC) $(SRC)readpk.c -o $(OBJ)readpk.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)readpk.c -o $(OBJ)readpk.o
 
 $(OBJ)readvf.o:  $(SRC)readvf.c $(OBJDEPEND)
-	$(CC) $(SRC)readvf.c -o $(OBJ)readvf.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)readvf.c -o $(OBJ)readvf.o
 
 $(OBJ)readtfm.o:  $(SRC)readtfm.c $(OBJDEPEND)
-	$(CC) $(SRC)readtfm.c -o $(OBJ)readtfm.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)readtfm.c -o $(OBJ)readtfm.o
 
 $(OBJ)drawit.o:  $(SRC)drawit.c $(OBJDEPEND)
-	$(CC) $(SRC)drawit.c -o $(OBJ)drawit.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)drawit.c -o $(OBJ)drawit.o
 
 $(OBJ)halfhyp.o:  $(SRC)halfhyp.c $(OBJDEPEND)
-	$(CC) $(SRC)halfhyp.c -o $(OBJ)halfhyp.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)halfhyp.c -o $(OBJ)halfhyp.o
 
 $(OBJ)epsfile.o:  $(SRC)epsfile.c $(OBJDEPEND)
-	$(CC) $(SRC)epsfile.c -o $(OBJ)epsfile.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)epsfile.c -o $(OBJ)epsfile.o
 
 $(OBJ)colors.o:  $(SRC)colors.c $(OBJDEPEND)
-	$(CC) $(SRC)colors.c -o $(OBJ)colors.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)colors.c -o $(OBJ)colors.o
 
 $(OBJ)writelx.o:  $(SSRC)writelx.c $(OBJDEPEND)
-	$(CC) $(WRITELXFLAGS) $(SSRC)writelx.c -o $(OBJ)writelx.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(WRITELXFLAGS) $(SSRC)writelx.c -o $(OBJ)writelx.o
 
 $(OBJ)subs.o:  $(SRC)subs.c $(OBJDEPEND)
-	$(CC) $(SRC)subs.c -o $(OBJ)subs.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)subs.c -o $(OBJ)subs.o
 
 $(OBJ)bookmks.o:  $(SRC)bookmks.c $(OBJDEPEND)
-	$(CC) $(SRC)bookmks.c -o $(OBJ)bookmks.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)bookmks.c -o $(OBJ)bookmks.o
 
 $(OBJ)rwconf.o:  $(SRC)rwconf.c $(OBJDEPEND)
-	$(CC) $(SRC)rwconf.c -o $(OBJ)rwconf.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)rwconf.c -o $(OBJ)rwconf.o
 
 $(OBJ)globals.o:  $(SRC)globals.c $(OBJDEPEND)
-	$(CC) $(SRC)globals.c -o $(OBJ)globals.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)globals.c -o $(OBJ)globals.o
 clean:
 	rm -f $(OBJ)*.o
 
diff -Nur tmview.orig/MakeSVGA tmview/MakeSVGA
--- tmview.orig/MakeSVGA	Sun Mar  4 14:57:22 2001
+++ tmview/MakeSVGA	Fri Nov  9 19:43:31 2001
@@ -13,7 +13,9 @@
 
 
 LL=gcc   -L /usr/lib
-CC=gcc -c -O2 -g -Wall $(WRITEVGAFLAGS) -I$(SRC) -I$(SSRC)
+CC=gcc -c $(WRITEVGAFLAGS)
+CFLAGS=-O2 -g -Wall
+INCLUDES=-I$(SRC) -I$(SSRC)
 
 
 OBJDEPEND=$(SSRC)defssvga.h $(SRC)defsgen.h $(SRC)subs.c
@@ -40,46 +42,46 @@
 
 
 $(OBJ)tmview.o:  $(SRC)tmview.c $(SRC)help.h $(OBJDEPEND)
-	$(CC) $(SRC)tmview.c -o $(OBJ)tmview.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)tmview.c -o $(OBJ)tmview.o
 
 $(OBJ)readdvi.o:  $(SRC)readdvi.c $(OBJDEPEND)
-	$(CC) $(SRC)readdvi.c -o $(OBJ)readdvi.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)readdvi.c -o $(OBJ)readdvi.o
 
 $(OBJ)readpk.o:  $(SRC)readpk.c $(OBJDEPEND)
-	$(CC) $(SRC)readpk.c -o $(OBJ)readpk.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)readpk.c -o $(OBJ)readpk.o
 
 $(OBJ)readvf.o:  $(SRC)readvf.c $(OBJDEPEND)
-	$(CC) $(SRC)readvf.c -o $(OBJ)readvf.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)readvf.c -o $(OBJ)readvf.o
 
 $(OBJ)readtfm.o:  $(SRC)readtfm.c $(OBJDEPEND)
-	$(CC) $(SRC)readtfm.c -o $(OBJ)readtfm.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)readtfm.c -o $(OBJ)readtfm.o
 
 $(OBJ)drawit.o:  $(SRC)drawit.c $(OBJDEPEND)
-	$(CC) $(SRC)drawit.c -o $(OBJ)drawit.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)drawit.c -o $(OBJ)drawit.o
 
 $(OBJ)halfhyp.o:  $(SRC)halfhyp.c $(OBJDEPEND)
-	$(CC) $(SRC)halfhyp.c -o $(OBJ)halfhyp.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)halfhyp.c -o $(OBJ)halfhyp.o
 
 $(OBJ)epsfile.o:  $(SRC)epsfile.c $(OBJDEPEND)
-	$(CC) $(SRC)epsfile.c -o $(OBJ)epsfile.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)epsfile.c -o $(OBJ)epsfile.o
 
 $(OBJ)colors.o:  $(SRC)colors.c $(OBJDEPEND)
-	$(CC) $(SRC)colors.c -o $(OBJ)colors.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)colors.c -o $(OBJ)colors.o
 
 $(OBJ)writevga.o:  $(SSRC)writevga.c $(OBJDEPEND)
-	$(CC) $(SSRC)writevga.c -o $(OBJ)writevga.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SSRC)writevga.c -o $(OBJ)writevga.o
 
 $(OBJ)subs.o:  $(SRC)subs.c $(OBJDEPEND)
-	$(CC) $(SRC)subs.c -o $(OBJ)subs.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)subs.c -o $(OBJ)subs.o
 
 $(OBJ)bookmks.o:  $(SRC)bookmks.c $(OBJDEPEND)
-	$(CC) $(SRC)bookmks.c -o $(OBJ)bookmks.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)bookmks.c -o $(OBJ)bookmks.o
 
 $(OBJ)rwconf.o:  $(SRC)rwconf.c $(OBJDEPEND)
-	$(CC) $(SRC)rwconf.c -o $(OBJ)rwconf.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)rwconf.c -o $(OBJ)rwconf.o
 
 $(OBJ)globals.o:  $(SRC)globals.c $(OBJDEPEND)
-	$(CC) $(SRC)globals.c -o $(OBJ)globals.o
+	$(CC) $(CFLAGS) $(INCLUDES) $(SRC)globals.c -o $(OBJ)globals.o
 clean:
 	rm -f $(OBJ)*.o
 
